### PR TITLE
docs: feedback shortcode

### LINKED ISSUE #1117: [docs] apply the "feedback" docs component to all pages
### Description

We currently handle the "feedback" section in two ways.

1. We have feedback component (docs/_includes/feedback.html) that is handles the feedback footer for all `elements` & `patterns` pages.

docs/elements/elements.html
```njk
 {% include 'feedback.html' %}
 ```

2. We add a block of HTML to each page that contains some classes and formatting for re-creating the feedback section.

docs/release-notes.njk
```
<div class="multi-column--min-300-wide section section--palette-default container feedback">
  <div>
    {% call getstarted.section("Feedback") -%}
      <p>To give feedback about anything on this page, <a href="mailto:digital-design-system@redhat.com">contact us</a>.</p>
    {%- endcall %}
  </div>
  <div>
    {% call getstarted.section("Other libraries") -%}
      <p>To learn more about our other libraries, visit <a href="../../get-started">this page</a>.</p>
    {%- endcall %}
  </div>
</div>
```

### Suggested solution

We should improve the feedback component logic so that it can work beyond the `elements` & `patterns` pages.

### Example API

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bennypowers**
closes #1117

## What I did

1. make a new `feedback` short code
2. use that new shortcode in the `feedback` include
3. use that new shortcode on the pages which have feedback divs

## Testing Instructions

[/about/how-we-build/](https://deploy-preview-1129--red-hat-design-system.netlify.app/about/how-we-build/) | [uxdot](https://ux.redhat.com/about/how-we-build/)
[/design-code-status/](https://deploy-preview-1129--red-hat-design-system.netlify.app/design-code-status/) | [uxdot](https://ux.redhat.com/design-code-status/)
[/elements/](https://deploy-preview-1129--red-hat-design-system.netlify.app/elements/) | [uxdot](https://ux.redhat.com/elements/)
[/elements/accordion/ (for example)](https://deploy-preview-1129--red-hat-design-system.netlify.app/elements/accordion/ (for example)) | [uxdot](https://ux.redhat.com/elements/accordion/ (for example))
[/foundations/color/](https://deploy-preview-1129--red-hat-design-system.netlify.app/foundations/color/) | [uxdot](https://ux.redhat.com/foundations/color/)
[/foundations/grid/](https://deploy-preview-1129--red-hat-design-system.netlify.app/foundations/grid/) | [uxdot](https://ux.redhat.com/foundations/grid/)
[/foundations/personalization/](https://deploy-preview-1129--red-hat-design-system.netlify.app/foundations/personalization/) | [uxdot](https://ux.redhat.com/foundations/personalization/)
[/foundations/spacing/](https://deploy-preview-1129--red-hat-design-system.netlify.app/foundations/spacing/) | [uxdot](https://ux.redhat.com/foundations/spacing/)
[/foundations/typography/](https://deploy-preview-1129--red-hat-design-system.netlify.app/foundations/typography/) | [uxdot](https://ux.redhat.com/foundations/typography/)
[/get-started/](https://deploy-preview-1129--red-hat-design-system.netlify.app/get-started/) | [uxdot](https://ux.redhat.com/get-started/)
[/get-started/design-system-kit/](https://deploy-preview-1129--red-hat-design-system.netlify.app/get-started/design-system-kit/) | [uxdot](https://ux.redhat.com/get-started/design-system-kit/)
[/get-started/fts-starter-kit/](https://deploy-preview-1129--red-hat-design-system.netlify.app/get-started/fts-starter-kit/) | [uxdot](https://ux.redhat.com/get-started/fts-starter-kit/)
[/get-started/icon-libraries/](https://deploy-preview-1129--red-hat-design-system.netlify.app/get-started/icon-libraries/) | [uxdot](https://ux.redhat.com/get-started/icon-libraries/)
[/get-started/product-logo-library/](https://deploy-preview-1129--red-hat-design-system.netlify.app/get-started/product-logo-library/) | [uxdot](https://ux.redhat.com/get-started/product-logo-library/)
[/patterns/](https://deploy-preview-1129--red-hat-design-system.netlify.app/patterns/) | [uxdot](https://ux.redhat.com/patterns/)
[/patterns/announcement/ (for example)](https://deploy-preview-1129--red-hat-design-system.netlify.app/patterns/announcement/ (for example)) | [uxdot](https://ux.redhat.com/patterns/announcement/ (for example))
[/release-notes/](https://deploy-preview-1129--red-hat-design-system.netlify.app/release-notes/) | [uxdot](https://ux.redhat.com/release-notes/)

### CODE REVIEW

**Comment by brianferry** [comment]
@coreyvickery - Is there a correct side for the "Feedback" header to show on?  On uxdot right now it seems to go on different sides for different pages?  

For instance, 

https://ux.redhat.com/about/how-we-build/

image
(On the left)

https://ux.redhat.com/patterns/

image
(On the right)

**Comment by bennypowers** [comment]
the switch to feedback on the right was done in patterns recently to match elements but the static content pages hadn't caught up yet, this PR fixes that by putting feedback on the right in all cases

File: docs/elements/index.md [hunk 1]
```diff
@@ -61,4 +61,7 @@ summaries:
   To request a new element or if updates need to be made to an existing element, <a href="https://github.com/RedHat-UX/red-hat-design-system/issues/new/choose" target="_blank">create a GitHub issue</a>.
 {% endsection %}
 
-{% include 'feedback.html' %}
+{% feedback %}
+  <h2>Elements</h2>
```
Comment (brianferry) [inline docs/elements/index.md:65]: Looks like this one should be "Patterns"

File: docs/get-started/icon-libraries.njk [hunk 2]
```diff
@@ -137,17 +137,7 @@ tags:
 
 {%- endcall %}
 
-<!-- Cross links -->
+{% feedback %}
+  <p>To learn more about how to use icons, visit our <a href="https://www.redhat.com/en/about/brand/standards/icons">brand standards page</a>.</p>
```
Comment (brianferry) [inline docs/get-started/icon-libraries.njk:141]: Looks like this needs the `<h2>Icons</h2>`

File: docs/patterns/index.md [hunk 3]
```diff
@@ -54,4 +54,7 @@ create uniform, accessible experiences.
 To request a new element or if updates need to be made to an existing element, 
 [contact us](mailto:digital-design-system@redhat.com).
 
-{% include 'feedback.html' %}
+{% feedback %}
+  <h2>Patterns</h2>
```
Comment (brianferry) [inline docs/patterns/index.md:58]: Should be `elements`?

File: docs/release-notes.njk [hunk 4]
```diff
@@ -421,13 +421,4 @@ title: Release notes
...
-
-<div class="multi-column--min-300-wide section section--palette-default container feedback">
-  {% call components.section("Feedback") -%}
-    <p>To give feedback about anything on this page, <a href="mailto:digital-design-system@redhat.com">contact us</a>.</p>
-  {%- endcall %}
-  {% call components.section("Design and code status") -%}
-    <p>To check the design and code status of a foundation or component, visit <a href="../design-code-status">this page</a>.</p>
-  {%- endcall %}
-</div>
+{% include 'feedback.html' %}
```
Comment (brianferry) [inline docs/release-notes.njk:424]: Needs the `Design and code status` added back in

File: docs/elements/index.md (hunk 1 above)
Comment (bennypowers) [reply docs/elements/index.md:65]: It's on the elements page though. @dcaryll do we want the feedback footer for elements to link to patterns and vice versa?

**Comment by coreyvickery** [comment]
@brianferry Right side is correct. Are we proposing to have the same heading and text across the entire website? Or can we create custom instances for certain sections?

**Comment by brianferry** [comment]
> @brianferry Right side is correct. Are we proposing to have the same heading and text across the entire website? Or can we create custom instances for certain sections?

If I get your meaning, the feedback section is the same between all the pages, but the other section (Other resources, patterns, elements, etc.) is something that can be customized between the sections of the website.

**Comment by coreyvickery** [comment]
> If I get your meaning, the feedback section is the same between all the pages, but the other section (Other resources, patterns, elements, etc.) is something that can be customized between the sections of the website.

@brianferry Yes, the `Feedback` section will say one thing and should be the same across all `Element` pages. The `Feedback` section can say another thing when used on all `Pattern` pages. The same goes for other sections; a custom message, but the same on all pages in that section. I hope that makes sense.

**Comment by coreyvickery** [comment]
> > > If I get your meaning, the feedback section is the same between all the pages, but the other section (Other resources, patterns, elements, etc.) is something that can be customized between the sections of the website.
> > 
> > 
> > @brianferry Yes, the `Feedback` section will say one thing and should be the same across all `Element` pages. The `Feedback` section can say another thing when used on all `Pattern` pages. The same goes for other sections; a custom message, but the same on all pages in that section. I hope that makes sense.
> 
> So for this PR we're adding code that will automatically generate the feedback section on any page that uses the shortcode. The feedback section will say
> 
> `To give feedback about anything on this page, [contact us](https://github.com/RedHat-UX/red-hat-design-system/discussions).` which seems fine to put on every page, right? Patterns doesn't need different content here than elements?
@brianferry I understand better now. Yes, that is fine to put on every page.

File: docs/patterns/index.md (hunk 3 above)
Comment (bennypowers) [reply docs/patterns/index.md:58]: see above @coreyvickery @dcaryll

Comment (bennypowers) [reply docs/patterns/index.md:52]: ping @coreyvickery

**Review by marionnegp — APPROVED** [review]
The Feedback section's showing up consistently on the right side of the page for me.

**Pull request merged**